ethereum_history_api/circuits/lib/src/verifiers/transaction_test.nr

```diff
@@ -60,3 +60,60 @@ mod assert_tx_proof {
         assert_tx_proof(proof_invalid_depth, transactions_root);
     }
 }
+
+mod assert_tx_equals {
+    use crate::verifiers::transaction::assert_tx_equals;
+    use crate::fixtures::mainnet::{
+        cancun::small_block::transaction::{transaction, encoded_tx, tx_type},
+        homestead::fork::transaction::{transaction as legacy_transaction, encoded_tx as legacy_encoded_tx, tx_type as legacy_tx_type}
+    };
+
+    #[test]
+    fn success() {
+        assert_tx_equals(tx_type, encoded_tx, transaction);
+    }
+
+    #[test]
+    fn legacy_tx() {
+        assert_tx_equals(legacy_tx_type, legacy_encoded_tx, legacy_transaction);
+    }
+
+    #[test(should_fail_with="Invalid tx type")]
+    fn wrong_tx_type() {
+        let wrong_tx_type = tx_type + 1;
+        assert_tx_equals(wrong_tx_type, encoded_tx, transaction);
+    }
+}
+
+mod verify_tx {
+    use crate::verifiers::transaction::verify_tx;
+    use crate::fixtures::mainnet::{
+        cancun::small_block::{
+        transaction::{transaction, encoded_tx, tx_type, tx_idx}, transaction_proof::proof,
+        header::transactions_root
+    },
+        homestead::fork::{
+        transaction::{
+        transaction as legacy_transaction, encoded_tx as legacy_encoded_tx, tx_type as legacy_tx_type,
+        tx_idx as legacy_tx_idx
+    },
+        transaction_proof::proof as legacy_proof, header::transactions_root as legacy_transactions_root
+    }
+    };
+
+    #[test]
+    fn success() {
+        verify_tx(tx_idx, tx_type, transaction, proof, transactions_root);
+    }
+
+    #[test]
+    fn legacy_tx() {
+        verify_tx(
+            legacy_tx_idx,
+            legacy_tx_type,
+            legacy_transaction,
+            legacy_proof,
+            legacy_transactions_root
+        );
+    }
```

Comment:
There is no fail test for invalid `tx_idx` (with message: `key does not match rlp-encoded transaction index`). I mean commenting follwing verifiers line
```rust
assert_tx_idx_equals(key_as_nibbles(tx_proof.key), tx_idx);
```
does not fail any test.